### PR TITLE
Added support for symfony/console v3

### PR DESCRIPTION
**Pull request by pkruithof**
See puli/discovery#19 for discussion

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -7,9 +7,15 @@ php:
...
+matrix:
+  include:
+    - php: 5.6
+      env: COMPOSER_FLAGS='--prefer-lowest --prefer-stable'
+  fast_finish: true
+
 before_script:
   - composer self-update
-  - composer install --dev --prefer-source
+  - composer update $COMPOSER_FLAGS --prefer-source -n
```
Comment (GrahamCampbell) [inline .travis.yml]: `-n` is not needed

Comment (GrahamCampbell) [inline .travis.yml]: travis already force enable it using environment variables

File: composer.json [hunk 2]
```diff
@@ -24,7 +24,7 @@
     },
     "require-dev": {
         "moontoast/math": "~1.1",
-        "symfony/console": "~2.3",
+        "symfony/console": "~2.3|^3.0",
```
Comment (GrahamCampbell) [inline composer.json]: should be `~` not `^`

File: composer.json [hunk 3]
```diff
@@ -24,7 +24,7 @@
...
         "doctrine/dbal": ">=2.3",
         "phpunit/phpunit": "~4.1",
```
Comment (GrahamCampbell) [inline composer.json]: would make sense to allow `~5.0` here too

File: .travis.yml (hunk 1 above)
Comment (pkruithof) [inline .travis.yml]: But it's non-interactive, composer could ask a question or something (maybe about GitHub tokens), if not now then maybe in a future release?

Comment (pkruithof) [inline .travis.yml]: Ah, nevermind, I did not know that. Are they using an environment variable or something?

Comment (GrahamCampbell) [inline .travis.yml]: > But it's non-interactive, composer could ask a question or something (maybe about GitHub tokens), if not now then maybe in a future release?

No, travis already force it into non-interactive mode.

Comment (ramsey) [inline .travis.yml]: I don't understand the use of "update" instead of "install."

**Comment by ramsey** [comment]
The tests are [passing here](https://travis-ci.org/ramsey/uuid/builds/97471220) because they're grabbing version 2.8.0 of symfony/console (and one is grabbing 2.3.0 for `--prefer-lowest`). The problem I mentioned in puli/discovery#19 is that the tests fail with errors when symfony/console 3.0 is used.

**Comment by pkruithof** [comment]
This build uses 3.0 and passes: https://travis-ci.org/ramsey/uuid/jobs/97472351

The failing build is because `--prefer-lowest` is not an option for `install`, but for `update`, so that's apparently the reason for `update` instead of `install`.

**Comment by ramsey** [comment]
All looks good. Thanks for your help!

**Pull request merged**